Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -383,7 +384,7 @@ function Publish-UpgradedServiceFabricApplication
         if ($UnregisterUnusedVersions)
         {
             Write-Host (Get-VstsLocString -Key SFSDK_UnregisterUnusedVersions)
-            foreach ($registeredAppTypes in Get-ServiceFabricApplicationTypeAction -ApplicationTypeName $names.ApplicationTypeName | Where-Object { $_.ApplicationTypeVersion -ne $names.ApplicationTypeVersion })
+            foreach ($registeredAppTypes in Get-ServiceFabricApplicationTypeAction -ApplicationTypeName $names.ApplicationTypeName)
```

Comment:
Just for the info, please let me know the reason to remove 'Where-Object' clause from the query

Reply:
$UnregisterUnusedVersions means that any unused type should be unregistered. We were excluding current version as it is being used in upgrade. However, the upgrade might fail and that means type is not in use. Hence, now we will try to unregister all types. If upgrade was successful, unregister to current type will fail - but that error will be supressed.